### ISSUE TITLE
sometimes oidc / oauth2 redirect URL port resets to 0 

### ISSUE THREAD

**Issue by butonic** [owncloud/client#9793]
### Pre-submission Checks

- [X] I [checked for similar issues](https://github.com/owncloud/client/issues?q=), but could not find any. I also checked the closed issues. I could not contribute additional information to any existing issue.
- [X] I will take the time to fill in all the required fields. I know that the bug report may be dismissed otherwise due to lack of information.

### Describe the bug

Toying around with oidc I clicked "reopen browser" but the url that was produced started with `http://127.0.0.1:0/?code=...` ... which won't work when the browser tries to redirect to it.

### Expected behavior

It should always use the random port of the localhost server.

### Steps to reproduce the issue

I tried to reproduce this by restarting the idp several times and wiping its account tokens and sessions ... but I feel it is caused by the clients local server timing out?

### Client version number

ownCloud 2.10.1 (build 7187) [82eefd](https://github.com/owncloud/client/commit/82eefdafa2ec7a1927df87c850d9e6f1e9dac6af) Apr 5 2022 04:25:21
Libraries Qt 5.12.11, OpenSSL 1.1.1k 25 Mar 2021
verwenden das Virtual Files plugin: wincfapi
windows-10.0.25136

### Installation path (Windows only)

C:\Program Files (x86)\ownCloud

### Server information

ocis master

**Comment by TheOneRing** [comment]
@butonic usually this happens if you opened multiple auth windows, only one will succeed.
If an error occurs during auth we will also restart the process so well start on a new port.

**Comment by TheOneRing** [comment]
@fmoc I think your currently looking into it

**Comment by fmoc** [comment]
Nope. I'm fixing the case where the `OAuth` object goes into an error state and the URL is reset to an empty string. But I'll have a look.

**Comment by fmoc** [comment]
See https://doc.qt.io/qt-6/qabstracthttpserver.html, we should replace our own implementation with a proper HTTP server. Obviously adds a dependency on #10701.
